src/Microsoft.DotNet.Darc/Darc/Operations/VirtualMonoRepo/VmrDiffOperation.cs

```diff
@@ -0,0 +1,343 @@
+            await AddRemoteAndGenerateDiff(tmpProductRepo, tmpVmrProductRepo, repo2.Branch, await GetDiffFilters(mapping));
+        }
+        finally
+        {
+            if (fileSystem.DirectoryExists(tmpPath))
+            {
+                var gitFiles = Directory.GetDirectories(tmpPath, GitDirectory, SearchOption.AllDirectories)
+                    .Select(gitDir => Path.Combine(gitDir, "objects"))
+                    .SelectMany(q => Directory.GetFiles(q, "*", SearchOption.AllDirectories));
+                foreach (var gitFile in gitFiles)
+                {
+                    var fileInfo = new FileInfo(gitFile);
+                    fileInfo.Attributes = FileAttributes.Normal;
+                    fileInfo.IsReadOnly = false;
+                }
+                fileSystem.DeleteDirectory(tmpPath, true);
+            }
+        }
+        return 0;
+    }
+
+    private async Task<string> PrepareVmr(DiffRepo vmr, string tmpPath, string mapping)
+    {
+        if (string.IsNullOrEmpty(mapping))
+        {
+            throw new ArgumentException($"When preparing VMR, mapping can't be null");
+        }
+
+        var vmrProductRepo = Path.Combine(tmpPath, Guid.NewGuid().ToString());
+        if (vmr.IsLocal)
+        {
+            await CheckoutBranch(vmr);
+            CopyDirectory(Path.Combine(vmr.Path, VmrInfo.SourceDirName, mapping), vmrProductRepo, true);
+        }
+        else
+        {
+            vmrProductRepo = await PartiallyCloneVmr(tmpPath, vmr, mapping);
+        }
+        await GitInitRepo(vmrProductRepo, vmr.Branch);
+
+        return vmrProductRepo;
+    }
+
+    private async Task<string> PrepareRepo(DiffRepo repo, string tmpPath)
+    {
+        string tmpProductRepo = Path.Combine(tmpPath, Path.GetFileName(repo.Path));
+
+        if (!repo.IsLocal)
+        {
+            await processManager.ExecuteGit(Path.GetDirectoryName(tmpProductRepo)!, [
+                "clone",
+                "--depth", "1",
+                repo.Path,
+                "-b", repo.Branch,
+                tmpProductRepo
+            ]);
+        }
+        else
+        {
+            await CheckoutBranch(repo);
+            CopyDirectory(repo.Path, tmpProductRepo, true);
+        }
+
+        return tmpProductRepo;
+    }
+
+    private async Task<(DiffRepo repo1, DiffRepo repo2)> ParseInput()
+    {
+        DiffRepo repo1, repo2;
+        var parts = options.Input.Split("..", StringSplitOptions.RemoveEmptyEntries);
+        if (parts.Length > 2 || parts.Length < 1)
+        {
+            throw new ArgumentException($"Invalid input {options.Input}");
+        }
+        
+        if (parts.Length == 1)
+        {
+            var currentPath = Directory.GetCurrentDirectory();
+            var res = await processManager.ExecuteGit(currentPath, "status");
+            res.ThrowIfFailed("Current directory is not a git repo");
+            repo1 = new DiffRepo(currentPath, string.Empty, true, IsLocalRepoVmr(currentPath));
```

Comment:
I would use `null` to signal no branch.

Reply:
or we could always set the branch and not allow things like repo..repo:branch

Reply:
I would not do that. I see the most common use case being just `darc vmr diff repo`

Reply:
but in that case we can just check the checked out branch/ref or assume main if it's a remote uri

Reply:
I don't follow..

1. If I have a working branch of VMR in one dir and a working branch of VMR in the other, I will just want to diff those two folders quickly.
2. Even if I had `main` of both, the `main` of VMR and `main` of the repo will not be always equal. Most likely there's a missing codeflow since the last changes. So if you want to check why the VMR is not building and how much is runtime behind (and see the diff), you still can run `darc vmr diff https://runtime..https://dotnet

Reply:
yup that's how I think the tool should work too. But if we do `darc vmr diff https://runtime..https://dotnet` you'll probably want to see the diff of the main branches, so I'm thinking when we're parsing the input, we set the branch to main even if it's missing. For local that's not really needed, we can go with null

Reply:
So let's set the default to:
- Checked out commit for local repos
- `main` for remote VMR
- `defaultRef` for remote repos

Alternatively, instead of `defaultRef` we could put the SHA of the repo in `source-manifest.json` when target repo is VMR and VMR SHA in Version.Details.xml when target is the repo

Reply:
modified the parsing so it handles things as we discussed